Sources/PackageConfig/Package.swift

```diff
@@ -100,12 +100,35 @@ enum Package {
 		let fileManager = FileManager.default
 
 		let swiftPMDir = swiftPath.replacingOccurrences(of: "bin/swiftc", with: "lib/swift/pm")
-		let versions = try! fileManager.contentsOfDirectory(atPath: swiftPMDir)
-		#warning("TODO: handle //swift-tools-version:4.2 declarations")
-		let latestSPM = versions.sorted().last!
-		let libraryPathSPM = swiftPMDir + "/" + latestSPM
+		let versions = try! fileManager.contentsOfDirectory(atPath: swiftPMDir).filter { $0 != "llbuild" }
+
+        let swiftToolsVersion = getSwiftToolsVersion()
+        let spmDir = versions.contains(swiftToolsVersion) ? swiftToolsVersion : versions.sorted().last!
+		let libraryPathSPM = swiftPMDir + "/" + spmDir
 
 		debugLog("Using SPM version: \(libraryPathSPM)")
 		return ["-L", libraryPathSPM, "-I", libraryPathSPM, "-lPackageDescription"]
 	}
+    
+    static private func getSwiftToolsVersion() -> String {
+        guard let data = FileManager.default.contents(atPath: "Package.swift") else {
```

Comment:
https://developer.apple.com/documentation/swift/string/3126738-init should allow you you to remove `let contents = String(data: data, encoding: .utf8)` and make it all in one line